jwst/dark_current/dark_current_step.py

```diff
@@ -32,13 +32,19 @@ def process(self, input):
                 result.meta.cal_step.dark = 'SKIPPED'
                 return result
 
-            # Open the dark ref file data model
-            dark_model = datamodels.DarkModel(self.dark_name)
+            detector = input_model.meta.instrument.detector
```

Comment:
Not a big deal at all, but because everything in the routine is instrument-dependent, rather than detector-dependent, you could just use the instrument name (input_model.meta.instrument.name). Then you also wouldn't have to worry about always doing the equality on just the first 3 characters of the string:

 instrument = input_model.meta.instrument.name
 if instrument == 'MIRI':
     ...